src/dune_pkg/rev_store.ml

```diff
@@ -100,26 +100,61 @@ let make_stderr () = Process.Io.make_stderr ~output_on_success:Swallow ~output_l
 (* to avoid Git translating its CLI *)
 let env = Env.add Env.initial ~var:"LC_ALL" ~value:"C"
 
+let handle_git_error git command dir fiber =
+  let stderr_path = Temp.create File ~prefix:"dune" ~suffix:"output" in
+  let stderr_to = Process.Io.file stderr_path Out in
+  let out =
+    Fiber.with_error_handler
```

Comment:
Doesn't this hide the original exception?

What about just changing `failure_mode` to just return the status and we can fail ourselves with a proper error.